src/components/dragLineLayer.ts

```diff
@@ -0,0 +1,224 @@
+///<reference path="../reference.ts" />
+
+module Plottable {
+
+export interface DragLineCallback<D> { (dragLineLayer: Components.DragLineLayer<D>): void; };
+
+export module Components {
+  export class DragLineLayer<D> extends GuideLineLayer<D> {
+    private _dragInteraction: Interactions.Drag;
+    private _detectionRadius = 3;
+    private _detectionEdge: d3.Selection<void>;
+    private _enabled = true;
+
+    private _dragStartCallbacks: Utils.CallbackSet<DragLineCallback<D>>;
+    private _dragCallbacks: Utils.CallbackSet<DragLineCallback<D>>;
+    private _dragEndCallbacks: Utils.CallbackSet<DragLineCallback<D>>;
+    private _disconnectInteraction: () => void;
+
+    constructor(orientation: string) {
+      super(orientation);
+      this.addClass("drag-line-layer");
+      this.addClass("enabled");
+
+      this._dragInteraction = new Plottable.Interactions.Drag();
+      this._dragInteraction.attachTo(this);
+
+      let grabbedLine = (p: Point) => {
+        return (this._isVertical() &&
+                 this.pixelPosition() - this.detectionRadius() <= p.x &&
+                 p.x <= this.pixelPosition() + this.detectionRadius()
+               ) ||
+               (
+                 !this._isVertical() &&
+                 this.pixelPosition() - this.detectionRadius() <= p.y &&
+                 p.y <= this.pixelPosition() + this.detectionRadius()
+               );
+      };
+
+      let dragging = false;
+      let interactionDragStartCallback = (start: Point) => {
+        if (grabbedLine(start)) {
+          dragging = true;
+          this._dragStartCallbacks.callCallbacks(this);
+        }
+      };
+      this._dragInteraction.onDragStart(interactionDragStartCallback);
+
+      let interactionDragCallback = (start: Point, end: Point) => {
+        if (dragging) {
+          this._setPixelPositionWithoutChangingMode(this._isVertical() ? end.x : end.y);
+          this._dragCallbacks.callCallbacks(this);
+        }
+      };
+      this._dragInteraction.onDrag(interactionDragCallback);
+
+      let interactionDragEndCallback = (start: Point, end: Point) => {
+        if (dragging) {
+          dragging = false;
+          this._dragEndCallbacks.callCallbacks(this);
+        }
+      };
+      this._dragInteraction.onDragEnd(interactionDragEndCallback);
+
+      this._disconnectInteraction = () => {
+        this._dragInteraction.offDragStart(interactionDragStartCallback);
+        this._dragInteraction.offDrag(interactionDragCallback);
+        this._dragInteraction.offDragEnd(interactionDragEndCallback);
+        this._dragInteraction.detachFrom(this);
+        delete this._disconnectInteraction;
```

Comment:
I still am struggling to see the value of deleting this value for 2 reasons.
* Do we want to error if a user were to call this function twice?
* An explicit deletion here tells future readers that there is a specific reason for deleting.  What is that specific reason?